### ISSUE TITLE
is feval taking callables of the form `f(y_true, y_pred)` or `f(y_pred, y_true)`?

### ISSUE THREAD

**Issue by startakovsky** [Microsoft/LightGBM#1180]
Hello, is `feval` taking callables of the form `f(y_true, y_pred)` or `f(y_pred, y_true)`?

See the following conflicting examples:
   - https://github.com/Microsoft/LightGBM/blob/master/examples/python-guide/advanced_example.py#L147
  -
 https://github.com/Microsoft/LightGBM/blob/07275da3d2d64ffb200befc101554b503e8f11ab/python-package/lightgbm/sklearn.py#L356

And in fact in `advanced_example.py` (the first bullet) above this seems correct, while the documentation states the opposite.

Am I missing something?

**Comment by startakovsky** [comment]
Or... is it just the case that it is different for the Scikit Learn API?

**Comment by guolinke** [comment]
for lgb.train or lgb.cv, you use `f(pred, eval_dataset)`.
for sklearn, you use ``func(label, pred)``, ``func(label, pred, weight)``
